protobuf-test/src/common/v2/test_serde_derive.rs

```diff
@@ -0,0 +1,29 @@
+use super::test_serde_derive_pb::*;
+use serde_json;
+use std::collections::HashMap;
+
+#[test]
+fn serialize_deserialize () {
```

Comment:
Could you please split this test into smaller tests, e. g. `test_enum`, `test_map`, `test_oneof` etc.

Reply:
Yup - mind if I split it up once I have everything working? Or is it hurting your reviewing at the moment (if so I'll split it up right now, no big deal)